breathecode/registry/models.py

```diff
@@ -496,12 +496,47 @@ def get_readme(self, parse=None, remove_frontmatter=False):
         return readme
 
     def parse(self, readme, format='markdown', remove_frontmatter=False):
+
+        readme = {
+            'clean': self.readme,
+            'decoded': Asset.decode(self.readme),
+            'raw': self.readme_raw,
+            'decoded_raw': Asset.decode(self.readme_raw)
+        }
+
+        if not isinstance(readme, dict):
+            readme = {}
+
+        if 'decoded' not in readme:
+            readme['decoded'] = Asset.decode(self.readme)
+
+        if 'clean' not in readme:
+            readme['clean'] = self.readme
+
+        if 'raw' not in readme:
+            readme['raw'] = self.readme_raw
+
+        if 'decoded_raw' not in readme:
+            readme['decoded_raw'] = Asset.decode(self.readme_raw)
+
         if format == 'markdown':
-            _data = frontmatter.loads(readme['decoded'])
-            readme['frontmatter'] = _data.metadata
-            readme['frontmatter']['format'] = format
-            readme['decoded'] = _data.content
-            readme['html'] = markdown.markdown(_data.content, extensions=['markdown.extensions.fenced_code'])
+            try:
+                _data = frontmatter.loads(readme['decoded'])
+                readme['frontmatter'] = _data.metadata
+                readme['frontmatter']['format'] = format
+                readme['decoded'] = _data.content
+                readme['html'] = markdown.markdown(_data.content,
+                                                   extensions=['markdown.extensions.fenced_code'])
+            except Exception:
+                readme['html'] = markdown.markdown(_data.content,
```

Comment:
This is wrong, if `frontmatter.loads` emit a exception, the variable `_data` will not exists, and this cause a exception in this line, in python use a variables that is undefined cause a exception